vent/coordinator/coordinator.py

```diff
@@ -3,65 +3,114 @@
 from typing import List, Dict
 
 import vent
-from vent.common.message import SensorValues, ControlSetting, Alarm, IPCMessageCommand
+from vent.common.message import SensorValues, ControlSetting, Alarm
 from vent.common.values import ValueName
+from vent.common import values
 from vent.common.message import SensorValueNew
-from vent.controller.control_module import get_control_module
-from vent.coordinator.ipc import IPC
+import vent.controller.control_module
+from vent.coordinator.ipc import IPC, IPCMessage, IPCCommand
 from vent.coordinator.process_manager import ProcessManager
 
 
 class CoordinatorBase:
     def __init__(self, sim_mode=False):
         # get_ui_control_module handles single_process flag
         # TODO: SHARED_ is a better prefix than COPY_, as not all fields are copy
-        self.control_module = get_control_module(sim_mode)
         self.COPY_sensor_values = None
         self.COPY_alarms = None
         self.COPY_active_alarms = {}
         self.COPY_logged_alarms = []
         self.COPY_control_settings = {}
         self.COPY_tentative_control_settings = {}
         self.COPY_last_message_timestamp = None
-        
-    def get_sensors(self) -> Dict[ValueName, SensorValueNew]:
-        # returns SensorValues struct
-        pass
+        self.lock = threading.Lock()
+        self._is_running = threading.Event()
 
-    # def get_alarms(self) -> List[Alarm]:
-    #     # returns list of Alarm structs
-    #     pass
+    def get_sensors(self) -> Dict[ValueName, SensorValueNew]:
+        with self.lock:
+            sensor_values = self.COPY_sensor_values
+        if sensor_values is None:
+            return None
+        # TODO: merge the API such that controller and coordinator use the same SensorValue class
```

Comment:
I agree. But if the tests don't break, I won't bother to defer the fixing in the next pull requests.

Reply:
:D

Reply:
well... the test don't break because i've been waiting for this PR to merge so i can merge my tests in... so this comment is effectively me telling you that yes it indeed does break the gui. sure we can merge this in and i'll PR my fix to this of that's how you'd rather do it.